components/QueryControls/index.tsx

```diff
@@ -0,0 +1,274 @@
+  });
+
+  const setNumberOfPosts = (newValue?: number) => {
+    setAttributes({
+      numberOfPosts: newValue,
+      posts: manualPosts.slice(0, newValue),
+    });
+  };
+
+  const setManualPost = (id: number, index: number) => {
+    const newManualPosts = [...manualPosts];
+    // If the post is already in the list, remove it.
+    if (id !== null && newManualPosts.includes(id)) {
+      newManualPosts.splice(newManualPosts.indexOf(id), 1, null);
+    }
+    newManualPosts.splice(index, 1, id);
+    setAttributes({ posts: newManualPosts });
+  };
+
+  return (
+    <>
+      <InspectorControls>
+        <PanelBody
+          title={__('Setup', 'wp-curate')}
+          initialOpen
+        >
+          {minNumberOfPosts !== undefined && minNumberOfPosts !== maxNumberOfPosts ? (
+            <RangeControl
+              label={__('Number of Posts', 'wp-curate')}
+              help={__('The maximum number of posts to show.', 'wp-curate')}
+              value={numberOfPosts}
+              onChange={setNumberOfPosts}
+              min={minNumberOfPosts}
+              max={maxNumberOfPosts}
+            />
+          ) : null}
+          <RangeControl
+            label={__('Offset', 'wp-curate')}
+            help={__('The number of posts to pass over.', 'wp-curate')}
+            onChange={(newValue) => setAttributes({ offset: newValue })}
+            value={offset}
+            min={0}
+            max={20}
+          />
+        </PanelBody>
+
+        <PanelBody
+          title={__('Select Posts', 'wp-curate')}
+          initialOpen={false}
+          className="manual-posts"
+        >
+          {manualPosts.map((_post, index) => (
+            <PanelRow
+              // eslint-disable-next-line react/no-array-index-key
+              key={index}
+              className={classnames(
+                'manual-posts__container',
+                { 'manual-posts__container--selected': manualPosts[index] },
+              )}
+            >
+              <span className="manual-posts__counter">{index + 1}</span>
+              <PostPicker
+                allowedTypes={allowedPostTypes.map((type) => type.slug)}
+                onReset={() => setManualPost(0, index)}
+                onUpdate={(id: number) => { setManualPost(id, index); }}
+                value={manualPosts[index] || 0}
+                className="manual-posts__picker"
+              />
+            </PanelRow>
+          ))}
+        </PanelBody>
+
+        <PanelBody
+          title={__('Query Parameters', 'wp-curate')}
+          initialOpen={false}
+        >
+          <Checkboxes
+            label={__('Post Types', 'wp-curate')}
+            value={postTypes}
+            onChange={(newValue) => setAttributes({ postTypes: newValue, backfillPosts: [] })}
+            options={displayTypes}
+          />
+          {allowedTaxonomies.map((taxonomy) => (
+            <Fragment key={taxonomy.slug}>
+              { /* @ts-ignore */ }
```

Comment:
What is TS complaining about here?

Reply:
@kevinfodness Term Selector has [default props](https://github.com/alleyinteractive/alley-scripts/blob/main/packages/block-editor-tools/src/components/term-selector/index.jsx#L42-L50), but somehow we're not communicating that properly and TS wants them included.